---
title: "Supplement Figure 1"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 3
    toc_float: true
    theme: flatly
    highlight: textmate
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
  
}
```


# Libraries

Load necessary libraries for analysis.

```{r Necessary Libraries, message=FALSE, warning=FALSE}
library(tidyverse)
library(readr)
library(readxl)
library()
```

Define my custom plot theme

```{r Custom plot theme, echo=TRUE, message=FALSE, warning=FALSE}

# Define the custom theme
# plot_theme_custom <- function() {
#   theme_minimal() +
#     theme(
#       #line for x and y axis
#       axis.line = element_line(linewidth = 1,
#                                color = "black"),
# 
#       #axis ticks only on x and y, length standard
#       axis.ticks.x = element_line(color = "black",
#                                 linewidth = 1),
#       axis.ticks.y = element_line(color = "black",
#                                 linewidth = 1),
#       axis.ticks.length = unit(0.05, "in"),
# 
#       #text and font
#       axis.text = element_text(color = "black",
#                                family = "Arial",
#                                size = 7),
#       axis.title = element_text(color = "black",
#                                 family = "Arial",
#                                 size = 9),
#       legend.text = element_text(color = "black",
#                                  family = "Arial",
#                                  size = 7),
#       legend.title = element_text(color = "black",
#                                   family = "Arial",
#                                   size = 9),
#       plot.title = element_text(color = "black",
#                                 family = "Arial",
#                                 size = 10),
# 
#       #blank background and border
#       panel.background = element_blank(),
#       panel.border = element_blank(),
# 
#       #gridlines for alignment
#       panel.grid.major = element_line(color = "grey80", linewidth = 0.5),  #grey major grid for align in illus
#       panel.grid.minor = element_line(color = "grey90", linewidth = 0.5) #grey minor grid for align in illus
#     )
# }

# saveRDS(plot_theme_custom, "data/plot_theme_custom.RDS")

theme_custom <- readRDS("data/plot_theme_custom.RDS")

```


Define a function to save plots as .pdf and .png

```{r pdf png saving function, echo=TRUE, message=FALSE, warning=FALSE}

save_plot <- function(plot, filename, 
                      folder = ".", 
                      width = 8, 
                      height = 6, 
                      units = "in", 
                      dpi = 300, 
                      add_date = TRUE) {
  if (missing(filename)) stop("Please provide a filename (without extension) for the plot.")

  date_str <- if (add_date) paste0("_", format(Sys.Date(), "%y%m%d")) else ""
  pdf_file <- file.path(folder, paste0(filename, date_str, ".pdf"))
  png_file <- file.path(folder, paste0(filename, date_str, ".png"))
  ggsave(filename = pdf_file, plot = plot, device = cairo_pdf, width = width, height = height, units = units, bg = "transparent")
  ggsave(filename = png_file, plot = plot, device = "png", width = width, height = height, units = units, dpi = dpi, bg = "transparent")
  message("Saved plot as Cairo PDF: ", pdf_file)
  message("Saved plot as PNG: ", png_file)
}

output_folder <- "C:/Users/emmap/OneDrive/Desktop/DXR Manuscript Materials/Fig pdfs"


#save plot function created
#to use: just define the plot name, filename_base, width, height
```

# Initial Dataframes

Create the initial dataframes from counts files needed to start analysis.

## Read in Counts Files from FeatureCounts

```{r Create My Dataframe With All Counts Files}
#load in libraries needed
#these counts files are from featureCounts, all saved as RDS objects

#prev Ind1 84-1 is now Ind4

# ####Individual 1 - 87-1####
# Counts_87_DOX_24 <- readRDS("data/QC/Counts_87_DOX_24.RDS")
# Counts_87_DMSO_24 <- readRDS("data/QC/Counts_87_DMSO_24.RDS")
# Counts_87_DOX_24rec <- readRDS("data/QC/Counts_87_DOX_24rec.RDS")
# Counts_87_DMSO_24rec <- readRDS("data/QC/Counts_87_DMSO_24rec.RDS")
# Counts_87_DOX_144rec <- readRDS("data/QC/Counts_87_DOX_144rec.RDS")
# Counts_87_DMSO_144rec <- readRDS("data/QC/Counts_87_DMSO_144rec.RDS")
# 
# ####Individual 2 - 78-1####
# Counts_78_DOX_24 <- readRDS("data/QC/Counts_78_DOX_24.RDS")
# Counts_78_DMSO_24 <- readRDS("data/QC/Counts_78_DMSO_24.RDS")
# Counts_78_DOX_24rec <- readRDS("data/QC/Counts_78_DOX_24rec.RDS")
# Counts_78_DMSO_24rec <- readRDS("data/QC/Counts_78_DMSO_24rec.RDS")
# Counts_78_DOX_144rec <- readRDS("data/QC/Counts_78_DOX_144rec.RDS")
# Counts_78_DMSO_144rec <- readRDS("data/QC/Counts_78_DMSO_144rec.RDS")
# 
# ####Individual 3 - 75-1####
# Counts_75_DOX_24 <- readRDS("data/QC/Counts_75_DOX_24.RDS")
# Counts_75_DMSO_24 <- readRDS("data/QC/Counts_75_DMSO_24.RDS")
# Counts_75_DOX_24rec <- readRDS("data/QC/Counts_75_DOX_24rec.RDS")
# Counts_75_DMSO_24rec <- readRDS("data/QC/Counts_75_DMSO_24rec.RDS")
# Counts_75_DOX_144rec <- readRDS("data/QC/Counts_75_DOX_144rec.RDS")
# Counts_75_DMSO_144rec <- readRDS("data/QC/Counts_75_DMSO_144rec.RDS")
# 
# ####Individual 4 - 84-1####
# Counts_84_DOX_24 <- readRDS("data/QC/Counts_84_DOX_24.RDS")
# Counts_84_DMSO_24 <- readRDS("data/QC/Counts_84_DMSO_24.RDS")
# Counts_84_DOX_24rec <- readRDS("data/QC/Counts_84_DOX_24rec.RDS")
# Counts_84_DMSO_24rec <- readRDS("data/QC/Counts_84_DMSO_24rec.RDS")
# Counts_84_DOX_144rec <- readRDS("data/QC/Counts_84_DOX_144rec.RDS")
# Counts_84_DMSO_144rec <- readRDS("data/QC/Counts_84_DMSO_144rec.RDS")
# 
# ####Individual 5 - 17-3####
# Counts_17_DOX_24 <- readRDS("data/QC/Counts_17_DOX_24.RDS")
# Counts_17_DMSO_24 <- readRDS("data/QC/Counts_17_DMSO_24.RDS")
# Counts_17_DOX_24rec <- readRDS("data/QC/Counts_17_DOX_24rec.RDS")
# Counts_17_DMSO_24rec <- readRDS("data/QC/Counts_17_DMSO_24rec.RDS")
# Counts_17_DOX_144rec <- readRDS("data/QC/Counts_17_DOX_144rec.RDS")
# Counts_17_DMSO_144rec <- readRDS("data/QC/Counts_17_DMSO_144rec.RDS")
# 
# ####Individual 6 - 90-1####
# Counts_90_DOX_24 <- readRDS("data/QC/Counts_90_DOX_24.RDS")
# Counts_90_DMSO_24 <- readRDS("data/QC/Counts_90_DMSO_24.RDS")
# Counts_90_DOX_24rec <- readRDS("data/QC/Counts_90_DOX_24rec.RDS")
# Counts_90_DMSO_24rec <- readRDS("data/QC/Counts_90_DMSO_24rec.RDS")
# Counts_90_DOX_144rec <- readRDS("data/QC/Counts_90_DOX_144rec.RDS")
# Counts_90_DMSO_144rec <- readRDS("data/QC/Counts_90_DMSO_144rec.RDS")
# 
# ####Individual 7 - 90-1REP####
# Counts_90REP_DOX_24 <- readRDS("data/QC/Counts_90REP_DOX_24.RDS")
# Counts_90REP_DMSO_24 <- readRDS("data/QC/Counts_90REP_DMSO_24.RDS")
# Counts_90REP_DOX_24rec <- readRDS("data/QC/Counts_90REP_DOX_24rec.RDS")
# Counts_90REP_DMSO_24rec <- readRDS("data/QC/Counts_90REP_DMSO_24rec.RDS")
# Counts_90REP_DOX_144rec <- readRDS("data/QC/Counts_90REP_DOX_144rec.RDS")
# Counts_90REP_DMSO_144rec <- readRDS("data/QC/Counts_90REP_DMSO_144rec.RDS")

```

## Create my Dataframe from Counts Files

Combine all of my counts files above into one dataframe, then convert to a matrix for further analysis.

```{r Put Counts Together Into Dataframe}
#change 84-1 to Ind4 now

# counts_raw_df <-
#    data.frame(
#      Counts_87_DOX_24,
#      Counts_87_DMSO_24$MCW_EMP_JT_R38_R1.bam,
#      Counts_87_DOX_24rec$MCW_EMP_JT_R39_R1.bam,
#      Counts_87_DMSO_24rec$MCW_EMP_JT_R41_R1.bam,
#      Counts_87_DOX_144rec$MCW_EMP_JT_R42_R1.bam,
#      Counts_87_DMSO_144rec$MCW_EMP_JT_R44_R1.bam,
#      Counts_78_DOX_24$MCW_EMP_JT_R45_R1.bam,
#      Counts_78_DMSO_24$MCW_EMP_JT_R47_R1.bam,
#      Counts_78_DOX_24rec$MCW_EMP_JT_R48_R1.bam,
#      Counts_78_DMSO_24rec$MCW_EMP_JT_R50_R1.bam,
#      Counts_78_DOX_144rec$MCW_EMP_JT_R51_R1.bam,
#      Counts_78_DMSO_144rec$MCW_EMP_JT_R53_R1.bam,
#      Counts_75_DOX_24$MCW_EMP_JT_R54_R1.bam,
#      Counts_75_DMSO_24$MCW_EMP_JT_R56_R1.bam,
#      Counts_75_DOX_24rec$MCW_EMP_JT_R57_R1.bam,
#      Counts_75_DMSO_24rec$MCW_EMP_JT_R59_R1.bam,
#      Counts_75_DOX_144rec$MCW_EMP_JT_R60_R1.bam,
#      Counts_75_DMSO_144rec$MCW_EMP_JT_R62_R1.bam,
#      Counts_84_DOX_24$MCW_EMP_JT_R27_R1.bam,
#      Counts_84_DMSO_24$MCW_EMP_JT_R29_R1.bam,
#      Counts_84_DOX_24rec$MCW_EMP_JT_R30_R1.bam,
#      Counts_84_DMSO_24rec$MCW_EMP_JT_R32_R1.bam,
#      Counts_84_DOX_144rec$MCW_EMP_JT_R33_R1.bam,
#      Counts_84_DMSO_144rec$MCW_EMP_JT_R35_R1.bam,
#      Counts_17_DOX_24$MCW_EMP_JT_R63_R1.bam,
#      Counts_17_DMSO_24$MCW_EMP_JT_R65_R1.bam,
#      Counts_17_DOX_24rec$MCW_EMP_JT_R66_R1.bam,
#      Counts_17_DMSO_24rec$MCW_EMP_JT_R68_R1.bam,
#      Counts_17_DOX_144rec$MCW_EMP_JT_R69_R1.bam,
#      Counts_17_DMSO_144rec$MCW_EMP_JT_R71_R1.bam,
#      Counts_90_DOX_24$MCW_EMP_JT_R72_R1.bam,
#      Counts_90_DMSO_24$MCW_EMP_JT_R74_R1.bam,
#      Counts_90_DOX_24rec$MCW_EMP_JT_R75_R1.bam,
#      Counts_90_DMSO_24rec$MCW_EMP_JT_R77_R1.bam,
#      Counts_90_DOX_144rec$MCW_EMP_JT_R78_R1.bam,
#      Counts_90_DMSO_144rec$MCW_EMP_JT_R80_R1.bam,
#      Counts_90REP_DOX_24$MCW_EMP_JT_R81_R1.bam,
#      Counts_90REP_DMSO_24$MCW_EMP_JT_R83_R1.bam,
#      Counts_90REP_DOX_24rec$MCW_EMP_JT_R84_R1.bam,
#      Counts_90REP_DMSO_24rec$MCW_EMP_JT_R86_R1.bam,
#      Counts_90REP_DOX_144rec$MCW_EMP_JT_R87_R1.bam,
#      Counts_90REP_DMSO_144rec$MCW_EMP_JT_R89_R1.bam
#    )

# saveRDS(counts_raw_df, "data/QC/counts_raw_df.RDS")

#now convert this into a matrix
# counts_raw_matrix <- counts_raw_df %>% column_to_rownames(var = "X") %>% as.matrix()

#write this to a csv so I can save it for later
# write.csv(counts_raw_matrix, "data/QC/counts_raw_matrix.csv")

#I also want to save this as an R object so I don't have to run the counts every time
# saveRDS(counts_raw_matrix, "data/QC/counts_raw_matrix.RDS")

counts_raw_matrix <- readRDS("data/QC/counts_raw_matrix.RDS")

dim(counts_raw_matrix)
#28395 is my initial amount of genes prior to filtering

```

## Factors and Colors

Include each of the factors for these experiments as well as their color schemes.

```{r Colors + Factors}

####Colors####

# txtime_col_old <- list(
#   "DOX_24T" = "#263CA5",
#   "VEH_24T" = "#444343",
#   "DOX_24R" = "#5B8FD1",
#   "VEH_24R" = "#757575",
#   "DOX_144R" = "#89C5E5",
#   "VEH_144R" = "#AAAAAA"
# )

#updated colors and factors:
ind_col <- list(
  "1" = "#FF9F64",
  "2" = "#78EBDA",
  "3" = "#ADCD77",
  "4" = "#E6CC50",
  "5" = "#A68AC0",
  "6" = "#F16F90",
  "6R" = "#C61E4E"
)

ind_col_nr <- list(
  "1" = "#FF9F64",
  "2" = "#78EBDA",
  "3" = "#ADCD77",
  "4" = "#E6CC50",
  "5" = "#A68AC0",
  "6" = "#F16F90"
)

time_col <- list(
  "t0" = "#005A4C",
  "t24" = "#328477",
  "t144" = "#8FB9B1")

tx_col <- list(
  "DOX" = "#499FBD", 
  "VEH" = "#BBBBBC")

txtime_col <- list(
  "DOX_t0" = "#263CA5",
  "VEH_t0" = "#444343",
  "DOX_t24" = "#5B8FD1",
  "VEH_t24" = "#757575",
  "DOX_t144" = "#89C5E5",
  "VEH_t144" = "#AAAAAA"
)
  
```


# QC Mapping Plots

Assess the alignment of each sample to hg38 as performed in command line previously.


## Set up initial dataframe 

```{r QC Alignment Setup}
#this dataframe contains my alignment percentages from featureCounts

#I originally had fC_DOXCounts which contained the individuals prior to their final order - here is the code that I used to change it

#I want to change these to reflect the final order of samples
# fC_DOXCounts <- fC_DOXCounts %>%
#   mutate(
#     Ind = case_when(
#       Line == "87-1" ~ "1",
#       Line == "78-1" ~ "2",
#       Line == "75-1" ~ "3",
#       Line == "84-1" ~ "4",
#       Line == "17-3" ~ "5",
#       Line == "90-1" ~ "6",
#       Line == "90-1REP" ~ "6R",
#       TRUE ~ NA_character_
#     ),
#   Tx = ifelse(Treatment == "DMSO", "VEH", "DOX"),
#   Timepoint = case_when(
#     Time == "24" ~ "t0",
#     Time == "24rec" ~ "t24",
#     Time == "144rec" ~ "t144",
#     TRUE ~ NA_character_
#                      ),
#     Condition = paste0(Tx, "_", Timepoint, "_", Ind)
#   )
# 
# #factor individual
# fC_DOXCounts <- fC_DOXCounts %>%
#   mutate(
#     Ind = factor(Ind, levels = c("1", "2", "3", "4", "5", "6", "6R"),
#                  ordered = TRUE)
#   )
# 
# 
# fC_DOXCounts <- fC_DOXCounts %>%
#   mutate(
#     Ind = factor(Ind, levels = c("1", "2", "3", "4", "5", "6", "6R"),
#                  ordered = TRUE),
#         Timepoint = factor(Timepoint, levels = c("t0", "t24", "t144"),
#                        ordered = TRUE),
#         Tx = factor(Tx, levels = c("DOX", "VEH"),
#                     ordered = TRUE)
#   ) %>%
#   mutate(
#     Condition = factor(
#       Condition,
#       levels = expand.grid(
#         Tx = levels(Tx),
#         Timepoint = levels(Timepoint),
#         Ind = levels(Ind)
#       ) %>%
#         transmute(Condition = paste(Tx, Timepoint, Ind, sep = "_")) %>%
#         pull(Condition),
#       ordered = TRUE
#     )
#   )

# saveRDS(fC_DOXCounts, "data/QC/featureCounts_DOXdata_updatedind.RDS")


fC_DOXCounts <- readRDS("data/QC/featureCounts_DOXdata_updatedind.RDS")

#Now I want to plot these values out
```


## Supplementary Figure 1A - Total Fragments by Sample

Assess the total number of fragments for each sample (paired-end reads)

```{r QC Mapping Plots Total Fragments}

####By Sample####
reads_by_sample <- c("DOX" = "#499FBD", "VEH" = "#BBBBBC")

fC_DOXCounts %>% 
  ggplot(., aes (x = Condition, 
                 y = Total_Align, 
                 fill = Tx, 
                 group_by = Ind))+
  geom_col()+
  geom_hline(aes(yintercept = 20000000))+
  scale_fill_manual(values = reads_by_sample)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_custom()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_DOXCounts %>% 
  ggplot(., aes (x = Tx, y= Total_Align, fill = Tx))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_DOXCounts %>% 
  ggplot(., aes (x =as.factor(Ind), y=Total_Align))+
  geom_boxplot(aes(fill=as.factor(Ind)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Timepoint####
reads_by_time <- c("t0" = "#263CA5",
                 "t24" = "#5B8FD1",
                 "t144" = "#89c5e5")

map_by_time <- fC_DOXCounts %>% 
  ggplot(., aes (x = Condition, y = Assigned_Align, 
                 fill = Timepoint, group_by = Ind))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_time)+
  ggtitle(expression("Total number of mapped reads by timepoint"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_custom()+
  theme(
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = "white"))

fC_DOXCounts$Time <- factor(fC_DOXCounts$Time, 
                            levels = c("t0", "t24", "t144"))


map_time_box <- fC_DOXCounts %>% 
  ggplot(., aes (x = Timepoint, y= Assigned_Align, fill = Timepoint))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_time)+
  ggtitle(expression("Total number of mapped reads by timepoint"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_custom()+
  theme(axis.ticks = element_line(linewidth = 1),
        axis.text.y = element_text(angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = "white"))

# save_plot(plot = map_time_box,
          # filename = "MappedReads_Boxplot_Supp1C_EMP_",
          # folder = output_folder)

####Total Mapped Reads Per Drug####

reads_by_sample <- c("DOX" = "#5B8FD1", "VEH" = "#BBBBBC")
map_sample_plot <- fC_DOXCounts %>% 
  ggplot(., aes (x = Condition, y = Assigned_Align, fill = Tx,
                 group_by = Ind))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_custom()+
  theme(
        axis.text.y = element_text(angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = "white"))

# save_plot(
#   plot = map_sample_plot,
#   filename = "MappedReads_Sample_Supp1A_EMP_",
#   folder = output_folder,
#   height = 5,
#   width = 8
# )

####Read Counts by Treatment####
map_tx_bp <- fC_DOXCounts %>% 
  ggplot(., aes (x =Tx, y= Assigned_Align, fill = Tx))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_custom()+
  theme(axis.text.y = element_text(angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = "white"))

# save_plot(plot = map_tx_bp,
#           filename = "MappedReads_Treatment_Supp1B_EMP_",
#           folder = output_folder)

####Total Reads Per Individual####
fC_DOXCounts %>% 
  ggplot(., aes (x =as.factor(Ind), y=Assigned_Align))+
  geom_boxplot(aes(fill=as.factor(Ind)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of mapped reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

```
